src/EditorFeatures/Core/Implementation/CodeActions/PreviewOperationWithTextView.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.CodeActions
+{
+    internal abstract class PreviewOperationWithTextView : CodeActionOperation
+    {
+        internal abstract Task<object> GetPreviewAsync(ITextView textView, CancellationToken cancellationToken);
```

Comment:
Following the pattern of CodeActionOperation. I think the reason is because we don't want to dictate what type this must be.  it can be a string, frameworkelement, and even more complicated guys that end up presenting complicated views.